core/iwasm/libraries/lib-wasi-threads/unit-test/test_tid_allocator.cpp

```diff
@@ -0,0 +1,68 @@
+/*
+ * Copyright (C) 2023 Amazon.com Inc. or its affiliates. All rights reserved.
+ * SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
+ */
+
+#include <gtest/gtest.h>
+
+#include "tid_allocator.h"
+
+#include <stdint.h>
+
+bool
+is_tid_valid(int32 tid)
+{
+    /* See: https://github.com/WebAssembly/wasi-threads#design-choice-thread-ids
+     */
+    return tid >= TID_MIN && tid <= TID_MAX;
+}
+
+TEST(TidAllocator, BasicTest)
+{
+    TidAllocator allocator;
```

Comment:
Worth using the test suite setup/teardown for the common allocator initialization/deinitialization?